Firestore/Example/Tests/Core/FSTQueryListenerTests.mm

```diff
@@ -427,97 +436,98 @@ - (void)testWillRaiseInitialEventWhenGoingOffline {
 
   DocumentViewChange change1{doc1, DocumentViewChange::Type::kAdded};
   DocumentViewChange change2{doc2, DocumentViewChange::Type::kAdded};
-  FSTViewSnapshot *expectedSnap1 = [[FSTViewSnapshot alloc]
-                initWithQuery:query
-                    documents:snap1.documents
-                 oldDocuments:[FSTDocumentSet documentSetWithComparator:snap1.query.comparator]
-              documentChanges:{change1}
-                    fromCache:YES
-                  mutatedKeys:snap1.mutatedKeys
-             syncStateChanged:YES
-      excludesMetadataChanges:YES];
-  FSTViewSnapshot *expectedSnap2 = [[FSTViewSnapshot alloc] initWithQuery:query
-                                                                documents:snap2.documents
-                                                             oldDocuments:snap1.documents
-                                                          documentChanges:{change2}
-                                                                fromCache:YES
-                                                              mutatedKeys:snap2.mutatedKeys
-                                                         syncStateChanged:NO
-                                                  excludesMetadataChanges:YES];
-  XCTAssertEqualObjects(events, (@[ expectedSnap1, expectedSnap2 ]));
+  ViewSnapshot expectedSnap1{
+      query,
+      /*documents=*/snap1.documents(),
+      /*old_documents=*/[FSTDocumentSet documentSetWithComparator : snap1.query().comparator],
+      /*document_changes=*/{ change1 },
+      snap1.mutated_keys(),
+      /*from_cache=*/true,
+      /*sync_state_changed=*/true,
+      /*excludes_metadata_changes=*/true};
+
+  ViewSnapshot expectedSnap2{query,
+                             /*documents=*/snap2.documents(),
+                             /*old_documents=*/snap1.documents(),
+                             /*document_changes=*/{change2},
+                             snap2.mutated_keys(),
+                             /*from_cache=*/true,
+                             /*sync_state_changed=*/false,
+                             /*excludes_metadata_changes=*/true};
+  XCTAssertTrue((events == std::vector<ViewSnapshot>{expectedSnap1, expectedSnap2}));
```

Comment:
This is very true. Yes, `XCTAssertEqualObjects` tries to call `isEqual:`, while `XCTAssertEqual` only works on primitives. I've been deliberately postponing porting tests properly just because it seems a lot of work for relatively little gain, compared to porting the main classes. I'd prioritize porting the core code for now and leave this as technical debt to revisit later (also, these tests would become easier to port once they're dealing exclusively with C++ classes).